syntax = "proto3";

package grpc;

message Nothing {}

message Map{
    string map = 1;
}

message PlayerInformation{
    string id = 1;
    int32 szx = 2;
    int32 szy = 3;
}

message GameInformation{
    int32 x = 1;
    int32 y = 2;
    string field = 3;
}

message Move{
    string id = 1;
    int32 move_x = 2;
    int32 move_y = 3;
}

message Id{
    string s = 1;
}

message Movement{
    int32 move_x = 1;
    int32 move_y = 2;
}

message Direction{
    string s = 1;
}

message PlayerMovement{
    string id = 1;
    int32 move_x = 2;
    int32 move_y = 3;
}

message PlayerTurn{
    string id = 1;
    string direction = 2;
}

service Game{
    rpc GetMap (Nothing) returns (Map);

    rpc GetPlayersMovements (Nothing) returns (stream PlayerMovement);
    rpc GetPlayersTurns (Nothing) returns (stream PlayerTurn);
    rpc GetPlayersShots (Nothing) returns (stream Id);

    rpc Move (Movement) returns (Nothing);
    rpc Turn (Direction) returns (Nothing);
    rpc Fire (Nothing) returns (Nothing);
}
